File: 02-backend/spring-boot-ecommerce/src/main/java/com/azet/ecommerce/service/CheckoutService.java
package com.azet.ecommerce.service;

import com.azet.ecommerce.dto.PaymentInfo;
import com.azet.ecommerce.dto.Purchase;
import com.azet.ecommerce.dto.PurchaseResponse;
import com.stripe.exception.StripeException;
import com.stripe.model.PaymentIntent;

public interface CheckoutService {

    PurchaseResponse placeOrder(Purchase purchase);

    PaymentIntent createPaymentIntent(PaymentInfo paymentInfo) throws StripeException;
}

File: 03-frontend/angular-ecommerce/src/app/services/checkout.service.ts
import { HttpClient } from '@angular/common/http';
import { Inject, Injectable, inject } from '@angular/core';
import { Purchase } from '../common/purchase';
import { Observable } from 'rxjs';
import { CartService } from './cart.service';
import { environment } from 'src/environments/environment';
import { PaymentInfo } from '../common/payment-info';

@Injectable({
  providedIn: 'root'
})
export class CheckoutService {

  // theCartService: CartService;
  
  storage: Storage = localStorage;

  private purchaseUrl = environment.luv2ShopApiUrl + "/checkout/purchase";

  private paymentIntentUrl = environment.luv2ShopApiUrl + "/checkout/payment-intent";

  constructor(private httpClient: HttpClient, cartService: CartService) {
    // this.theCartService = cartService;
   }

  placeOrder(purchase: Purchase): Observable<any> {
    // this.theCartService.cartItems = [];
    this.storage.clear();
    
    return this.httpClient.post<Purchase>(this.purchaseUrl, purchase);
  }

  createPaymentIntent(paymentInfo: PaymentInfo): Observable<any> {

    return this.httpClient.post<PaymentInfo>(this.paymentIntentUrl, paymentInfo);
  }
}
